public/app/plugins/datasource/elasticsearch/query_builder.js

```diff
@@ -243,19 +242,26 @@ function (queryDef) {
         }
       });
     }
-
+    var size = 500;
+    if(queryDef.size){
+      if(this.esVersion >= 5 && queryDef.size !== 0) {
```

Comment:
ES 5.x doesn't support `"size": 0`. So it's more of a check not to break anything or else we could throw an exception in that case and let the user fix it.